OpenRA.Mods.Common/Projectiles/Bullet.cs

```diff
@@ -286,8 +285,11 @@ public IEnumerable<IRenderable> Render(WorldRenderer wr)
 				{
 					var dat = world.Map.DistanceAboveTerrain(pos);
 					var shadowPos = pos - new WVec(0, 0, dat.Length);
-					foreach (var r in anim.Render(shadowPos, wr.Palette(info.ShadowPalette)))
-						yield return r;
+					var shadow = new float3(info.ShadowColor.R, info.ShadowColor.G, info.ShadowColor.B) / 255f;
```

Comment:
Wouldn't it make sense to cache this in the `ctor` to avoid allocating a new float3 every render frame?

Reply:
I agree.

> avoid allocating a new float3 every render frame

And a DIV (or even multiple ones, since it's 3 values?), fwiw.

Reply:
Done.